### DISCUSSION TITLE
How is this possible?

### DISCUSSION THREAD

**Discussion by AaronPhilipp** [jdepoix/youtube-transcript-api#171] [Q&A] [answered]
Thank you first for your great work! The API is really helpful! What I would like to know is, how the module works? When I would like to get the captions with the youtube data API I need OAuth and I'm restricted because of the quota limit. How do you access the captions from the video?

Thank you in advance,

Aaron.

**Comment by jdepoix** [comment, marked as answer]
Hi @AaronPhilipp,
Thank you for the kind words! 😊 
The main idea of the module is to emulate the HTTP calls which are executed by YouTube's web client. Since you are able to enable a video's subtitles even if you are not logged in, we can call those endpoints unauthenticated as well. 
If you are curious about the details, just have a look at the code (it really isn't that complicated :wink:)! You'll find the main logic in `youtube_transcript_api._transcripts.py` and it only is around 300 lines of code.
